### PR TITLE
remove amp autocast default cuda setting.

### PR DESCRIPTION
**Pull request by lkk12014402**
remove amp autocast default cuda setting

### CODE REVIEW

**Review by WeiweiZhang1 — APPROVED** [review]

**Comment by wenhuach21** [comment]
please make sure "cuda:3" and "hpu:4" could work

**Comment by lkk12014402** [comment]
> please make sure "cuda:3" and "hpu:4" could work

add new variable self.amp_device_type

File: auto_round/autoround.py [hunk 1]
```diff
@@ -876,10 +884,16 @@ def quant_block(self, block, input_ids, input_others, q_input=None, device=torch
...
+                    current_input_ids,
+                    current_input_others,
+                    self.amp,
+                    self.amp_dtype,
+                    self.amp_device_type,
+                    device,
                 )
                 if self.amp and not check_is_cpu(device):
-                    with autocast(device_type="cuda", dtype=self.amp_dtype):
+                    with autocast(device_type=self.amp_device_type, dtype=self.amp_dtype):
```
Comment (wenhuach21) [inline auto_round/autoround.py:896]: a better way is just detect whether ":" in device string, and truncate the string, as there may more device like xpu, npu, etc

Comment (lkk12014402) [reply auto_round/autoround.py:896]: As I know, some code do not check npu or xpu device and we don't validate such devices. If we want to adapt other device, we need validate it and revise code.

**Comment by wenhuach21** [comment]
please review pr https://github.com/intel/auto-round/pull/64

**Pull request closed without merging**